src/xds/metrics.rs

```diff
@@ -58,3 +85,18 @@ impl Recorder<ConnectionTerminationReason, u64> for Metrics {
             .inc_by(count);
     }
 }
+
+impl Recorder<DeltaDiscoveryResponse, ()> for Metrics {
+    fn record(&self, response: &DeltaDiscoveryResponse, _: ()) {
+        let type_url = TypeUrl {
+            url: response.type_url.clone(),
```

Comment:
Feels like there should be a way of doing this without cloning

Reply:
Also, should I be getting the type_url from the response, or for each resource in response.resources?